grype/db/v5/matcher/java/matcher_integration_test.go

```diff
@@ -0,0 +1,689 @@
+//go:build integration
+// +build integration
+
+package java
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+)
+
+// TestMavenSearch_GetMavenPackageBySha tests the GetMavenPackageBySha method of the MavenSearch struct.
+// This is an integration test and requires network access to search.maven.org.
+// It is not intended to be run as part of the normal test suite.
+// Use this to validate rate limiting in [maven_search.go] and the ability to fetch package data from maven.org.
+func TestMavenSearch_GetMavenPackageBySha(t *testing.T) {
+	ctx := context.Background()
+
+	ms := newMavenSearch(http.DefaultClient, "https://search.maven.org/solrsearch/select")
+
+	// Known SHA1s to test with, using a large number of known good SHA1s to validate rate limiting
```

Comment:
I don't think we should be hitting maven in this way during a test that could be run by multiple devs at a time and by automation. I'm not against leaving the test in if we add a build tag that isn't on the common path (run by default or in CI). In that way if we run into issues again we can enable the test for debugging purposes.

Reply:
I think adding a small unit test that ensures that the rate limiter is wired up by default when using `mavenSearch` would be a good enough sanity check.

Reply:
Doh - I'd originally had
```
//go:build integration
// +build integration
```
At the top but the IDE skips compilation errors with that so been removing it, will add back if that's good enough, totally agree this shouldn't be a unit test :) 

On that, would adding the tags above skip the tests in your CI?

Reply:
ah, the tag is there and we're not using tags to select integration tests, so it wouldn't be selected today... maybe let's change the tag to something that doesn't look like it overlaps with the average integration test, something like `stress` or `api-limits` as a build tag.

Reply:
sounds good!